dev/NumberBox/NumberBox.cpp

```diff
@@ -46,6 +46,8 @@ NumberBox::NumberBox()
 
     SetDefaultStyleKey(this);
     SetDefaultInputScope();
+
+    this->RegisterPropertyChangedCallback(winrt::AutomationProperties::NameProperty(), { this , &NumberBox::OnAutomationPropertiesNamePropertyChanged });
```

Comment:
It doesn't look like this is ever unregistered, which would lead to memory leaks.  Please use [RegisterPropertyChanged](https://github.com/microsoft/microsoft-ui-xaml/blob/513ed8267007c8e0b3055c6b924e2c2c58601503/dev/inc/CppWinRTHelpers.h#L216) to get a revoker object for this registration.

Reply:
I though since we are registered to the NumberBox itsself, this wouldn't be necessary as both should have the same lifecycle.

Reply:
Ah.. hmm. I think you're right. @jevansaks or @ranjeshj could you confirm please :)

Reply:
Yes that's technically correct.

Reply:
The best kind of correct?

Jokes aside, I'll add a comment to point that out so this won't cause confusion down the line.